torch/_export/wrappers.py

```diff
@@ -115,3 +126,96 @@ def call(self, *args):
 
     cls.__call__ = call
     return cls
+
+
+def _register_and_get_spec_proxy_in_tracer(tracer, name, spec):
+    fx_name = name + "0"
+    if hasattr(tracer.root, fx_name):
+        assert getattr(tracer.root, fx_name) == spec
+        return tracer.create_proxy("get_attr", fx_name, (), {})
+
+    qualname = tracer.get_fresh_qualname(name)
+    setattr(tracer.root, qualname, spec)
+    return tracer.create_proxy("get_attr", qualname, (), {})
+
+
+def _mark_subclass_constructor_exportable_experimental(constructor_subclass):
+    """
+    Experimental decorator that makes subclass to be traceable in export
+    with pre-dispatch IR. To make your subclass traceble in export, you need to:
+        1. Implement __init__ method for your subclass (Look at DTensor implementation)
+        2. Decorate your __init__ method with _mark_constructor_exportable_experimental
+        3. Put torch._dynamo_disable decorator to prevent dynamo from peeking into its' impl
+
+    Example:
+
+    class FooTensor(torch.Tensor):
+        @staticmethod
+        def __new__(cls, elem, *, requires_grad=False):
+            # ...
+            return torch.Tensor._make_subclass(cls, elem, requires_grad=requires_grad)
+            
+        @torch._dynamo_disable 
+        @_mark_subclass_constructor_exportable_experimental
+        def __init__(self, elem, ...):
+            # ...
+    """
+
+    def wrapper(*args, **kwargs):
+        assert is_traceable_wrapper_subclass_type(
+            type(args[0])
+        ), "Can only apply _mark_constructor_exportable_experimental on tensor subclass constructors."
+        assert callable(constructor_subclass) and (
+            constructor_subclass.__name__ == "__init__"
+        ), "_mark_constructor_exportable_experimental can only be applied on tensor subclass constructors."  # noqa: B950
+        constructor_subclass(*args, **kwargs)
+        if not torch._C._is_torch_function_mode_enabled():
+            return
+        torch_function_mode_stack = torch.overrides._get_current_function_mode_stack()
```

Comment:
you still have a nested indent that you can fix here 😛 I would do something like:
```
        if not torch._C._is_torch_function_mode_enabled():
            return
       filtered_modes = [m for m in torch.overrides._get_current_function_mode_stack() if isinstance(m, PreDispatchTorchFunctionMode)]
       assert len(filtered_modes) <= 1, "duplicated pre-dispatch mode found"
       if len(filtered_modes) == 0:
           return
       predispatch_mode = filtered_modes[0]
       # regular logic...
```